R/tbl_baseline_chg.R

```diff
@@ -0,0 +1,259 @@
+      names_sort = TRUE
+    ) |>
+    # add in denominator for the header Ns
+    dplyr::right_join(
+      gtsummary::select(denominator, all_of(c(id, by))),
+      by = c(id, by),
+      relationship = "many-to-one"
+    )
+
+  # Build results tables ----------------------------------------------------
+  # Summary of AVAL
+  tbl_aval <-
+    df_change_baseline |>
+    dplyr::select(all_of(by), starts_with(analysis_variable)) |>
+    dplyr::rename_with(~ str_remove(., paste0("^", analysis_variable, "_"))) %>%
+    # after reshape all column labels are the same, so changing them to the variable name
+    labelled::remove_var_label() |>
+    tbl_roche_summary(
+      by = any_of(by),
+      nonmissing = "always", # include the non-missing count in summary
+      # round mean/sd/median/min/max,
+      type = everything() ~ "continuous2",
+      digits = list(
+        all_continuous() ~ list(
+          mean = 2, sd = 2, median = 2,
+          min = 1, max = 1
+        )
+      )
+    )
+
+  # Building a table change values at each visit
+  tbl_chg <-
+    df_change_baseline |>
+    dplyr::select(all_of(by), starts_with(change_variable)) |>
+    dplyr::rename_with(~ str_remove(., paste0("^", change_variable, "_"))) %>%
+    # after reshape all column labels are the same, so changing them to the variable name
+    labelled::remove_var_label() |>
+    # using `tbl_roche_summary()` as the default continuous variable summary matches our spec
+    tbl_roche_summary(
+      by = any_of(by),
+      nonmissing = "always", # include the non-missing count in summary
+      # round mean/sd/median/min/max
+      type = everything() ~ "continuous2",
+      digits =
+        list(
+          all_continuous() ~ list(
+            mean = 2, sd = 2, median = 2,
+            min = 1, max = 1
+          )
+        ),
+      include = everything() & !all_of(baseline_level) # Remove the baseline visit from summary
+    )
+
+  # Merge tables together
+  baseline_chg_tbl <-
+    list(tbl_aval, tbl_chg) |>
+    gtsummary::tbl_merge(tab_spanner = FALSE) |>
+    gtsummary::modify_spanning_header(gtsummary::all_stat_cols() ~ header_label) |>
+    gtsummary::modify_header(
+      gtsummary::all_stat_cols() & ends_with("_1") ~ "Value at Visit",
+      gtsummary::all_stat_cols() & ends_with("_2") ~ "Change from Baseline",
+      label = "Visit"
+    ) |>
+    # sort the stat columns together within treatment group
+    gtsummary::modify_table_body(
+      \(.x) {
+        stat_cols <- dplyr::select(.x, gtsummary::all_stat_cols()) |>
+          names() |>
+          sort()
+        dplyr::relocate(.x, all_of(stat_cols), .after = "label")
+      }
+    )
+
+  # return tbl -----------------------------------------------------------------
+  baseline_chg_tbl[["call_list"]] <- list(tbl_baseline_chg = match.call())
+  baseline_chg_tbl$inputs <- tbl_baseline_inputs
+  baseline_chg_tbl |>
+    structure(class = c("tbl_baseline_chg", "gtsummary"))
```

Comment:
I think we need to keep the merge class, or `gather_ard()` don't work, right? Or you'll need to gather them yourself and add them to `.$cards` (I think that is the name, but can you double check if you use this method?) Can you add a test with `gather_ard()`?

Reply:
great point! I've added the code you suggested to add the ard objects to the class